src/mlib/int128.h

```diff
@@ -0,0 +1,614 @@
+#ifndef MLIB_INT128_H_INCLUDED
+#define MLIB_INT128_H_INCLUDED
+
+#include "./macros.h"
+
+#include <stdbool.h>
+#include <inttypes.h>
+#include <string.h>
+#include <stdlib.h>
+
+MLIB_C_LINKAGE_BEGIN
+
+/**
+ * @brief A 128-bit binary integer
+ */
+typedef union {
+   struct {
+      uint64_t lo;
+      uint64_t hi;
+   } r;
+#if defined(__SIZEOF_INT128__)
+   // These union members are only for the purpose of debugging visualization
+   // and testing, and will only appear correctly on little-endian platforms.
+   __int128_t signed_;
+   __uint128_t unsigned_;
+#endif
```

Comment:
Accessing an inactive union data member is UB according to the C++ standard. Can these be omitted in favor of bit-cast (memcpy) to/from `__(u)int128_t` where applicable (only in `int128.test.cpp`)? e.g.:
```cpp
#ifdef __SIZEOF_INT128__
      // When we have an existing i128 impl, test against that:
      __uint128_t num_u128;
      memcpy (&num_u128, &num, sizeof (__uint128_t));

      __uint128_t denom_u128;
      memcpy (&denom_u128, &denom, sizeof (__uint128_t));

      __uint128_t q_u128 = num_u128 / denom_u128;
      __uint128_t r_u128 = num_u128 % denom_u128;

      mlib_int128 quotient;
      memcpy (&quotient, &q_u128, sizeof (__uint128_t));

      mlib_int128 remainder;
      memcpy (&remainder, &r_u128, sizeof (__uint128_t));

      CHECK (q.quotient == quotient);
      CHECK (q.remainder == remainder);
#endif
```
This would also reduce complexity/noise for `mlib_u128` (can be made a simple struct).

Reply:
I think the `memcpy`/`bit_cast` is a good idea in the test cases. I prefer to leave the union members because they have been life-saving while debugging.